code/sonalyze/daemon/perform.go

```diff
@@ -0,0 +1,387 @@
+		fmt.Fprintf(stderr, "FATAL ERROR: Failing to open logger: %s", err.Error())
+		os.Exit(1)
+	}
+
+	// Note "daemon" is not a command here
+	http.HandleFunc("/add", httpAddHandler(dc))
+	http.HandleFunc("/jobs", httpGetHandler(dc, "jobs"))
+	http.HandleFunc("/load", httpGetHandler(dc, "load"))
+	http.HandleFunc("/uptime", httpGetHandler(dc, "uptime"))
+	http.HandleFunc("/profile", httpGetHandler(dc, "profile"))
+	http.HandleFunc("/parse", httpGetHandler(dc, "parse"))
+	http.HandleFunc("/metadata", httpGetHandler(dc, "metadata"))
+	// These request names are compatible with the older `infiltrate` and with the upload infra
+	// already running on the clusters.
+	http.HandleFunc("/sonar-freecsv", httpPostHandler(dc, "sample", "text/csv"))
+	http.HandleFunc("/sysinfo", httpPostHandler(dc, "sysinfo", "application/json"))
+
+	var programFailed bool
+	s := httpsrv.New(dc.Verbose, int(dc.port), func(err error) {
+		programFailed = true
+	})
+	go s.Start()
+
+	// Wait here until we're stopped by SIGHUP (manual) or SIGTERM (from OS during shutdown).
+	//
+	// TODO: IMPROVEME: For SIGHUP, we should not exit but should instead reread the password file
+	// and the cluster aliases file.
+	process.WaitForSignal(syscall.SIGHUP, syscall.SIGTERM)
+	s.Stop()
+
+	if programFailed {
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+// HTTP handlers
+//
+// Documented behavior: the server will close the request body, we don't need to do it.
+//
+// I can find no documentation about needing to consume the body in case of an early (error)
+// return, nor anything obvious in the net/http source code to indicate this, nor has google
+// turned up anything.  So request handler code assumes it's not necessary.
+
+func httpGetHandler(
+	dc *DaemonCommand,
+	command string,
+) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		_, _, clusterName, ok := requestPreamble(dc, w, r, "GET", dc.getAuthenticator, authRealm, "")
+		if !ok {
+			return
+		}
+
+		verb := command
+		arguments := []string{
+			"--data-path",
+			path.Join(dc.jobanalyzerDir, "data", clusterName),
+		}
+		for name, vs := range r.URL.Query() {
+			if name == "cluster" {
+				continue
+			}
+
+			if !argOk(command, name) {
+				w.WriteHeader(400)
+				fmt.Fprintf(w, "Bad parameter %s", name)
+				if dc.Verbose {
+					statusWarningf("Bad parameter %s", name)
+				}
+				return
+			}
+
+			// Repeats are OK, sonalyze allows them in a number of cases.  Booleans carry the magic
+			// boolean value, allowing us to construct a boolean option without a value without
+			// tracking name->type mappings.  This is a hack, but it works.  The reason we have to
+			// exclude the value for boolean options (`--some-gpu` instead of `--some-gpu=true`) is
+			// a limitation of Rust's `clap` library.  The reason the value carried is not simply
+			// "true" is that that is a more likely value for some other parameters (host names?)
+			// and we can't exclude it here without risk.
```

Comment:
Add a TODO here now that we're not dependent on the Rust-based sonalyze, and/or file a bug about this.